Content.Server/StationEvents/Events/RandomSentienceRule.cs

```diff
@@ -10,24 +10,42 @@ public sealed class RandomSentienceRule : StationEventSystem<RandomSentienceRule
 {
     protected override void Started(EntityUid uid, RandomSentienceRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
-        HashSet<EntityUid> stationsToNotify = new();
+        if (!TryGetRandomStation(out var station))
+            return;
 
         var targetList = new List<Entity<SentienceTargetComponent>>();
-        var query = EntityQueryEnumerator<SentienceTargetComponent>();
-        while (query.MoveNext(out var targetUid, out var target))
+        var query = EntityQueryEnumerator<SentienceTargetComponent, TransformComponent>();
+        while (query.MoveNext(out var targetUid, out var target, out var xform))
         {
+            if (StationSystem.GetOwningStation(targetUid, xform) != station)
+                continue;
+
             targetList.Add((targetUid, target));
         }
 
-        RobustRandom.Shuffle(targetList);
-
         var toMakeSentient = RobustRandom.Next(2, 5);
```

Comment:
This isn't your fault, but these magic numbers really should be moved to DataFields on the rule component.